config/boards/ayn-odin2.conf

```diff
@@ -0,0 +1,87 @@
+declare -g EXTRAWIFI="no"
+declare -g BOOTCONFIG="none"
+declare -g IMAGE_PARTITION_TABLE="gpt"
+declare -g -a ABL_DTB_LIST=("qcs8550-ayn-odin2-hypdtbo")
+
+# Use the full firmware, complete linux-firmware plus Armbian's
+declare -g BOARD_FIRMWARE_INSTALL="-full"
+
+function post_family_config_branch_sm8550__pmos_kernel() {	
+	declare -g KERNELSOURCE='https://github.com/edk2-porting/linux-next'
+	declare -g KERNEL_MAJOR_MINOR="6.7" # Major and minor versions of this kernel.
+	declare -g KERNELBRANCH="branch:work/sakuramist-odin2"
+	# declare -g LINUXCONFIG="odin2_defconfig"
+	declare -g LINUXCONFIG="linux-${ARCH}-${BRANCH}" # for this board: linux-arm64-sm8550
+	display_alert "Setting up kernel ${KERNEL_MAJOR_MINOR} for" "${BOARD}" "info"
+}
+
+function ayn-odin2_is_userspace_supported() {
+	[[ "${RELEASE}" == "jammy" ]] && return 0
+	[[ "${RELEASE}" == "trixie" ]] && return 0
+	[[ "${RELEASE}" == "noble" ]] && return 0
+	return 1
+}
+
+function post_family_tweaks__ayn-odin2_enable_services() {
+	if ! ayn-odin2_is_userspace_supported; then
+		if [[ "${RELEASE}" != "" ]]; then
+			display_alert "Missing userspace for ${BOARD}" "${RELEASE} does not have the userspace necessary to support the ${BOARD}" "warn"
+		fi
+		return 0
+	fi
+
+	# We need unudhcpd from armbian repo, so enable it
+	mv "${SDCARD}"/etc/apt/sources.list.d/armbian.list.disabled "${SDCARD}"/etc/apt/sources.list.d/armbian.list
+
+	do_with_retries 3 chroot_sdcard_apt_get_update
+	display_alert "$BOARD" "Installing board tweaks" "info"
+	do_with_retries 3 chroot_sdcard_apt_get_install alsa-ucm-conf unudhcpd mkbootimg git make gcc g++ liblzma-dev zlib1g-dev meson cmake
+
+	# Disable armbian repo back
+	mv "${SDCARD}"/etc/apt/sources.list.d/armbian.list "${SDCARD}"/etc/apt/sources.list.d/armbian.list.disabled
+	do_with_retries 3 chroot_sdcard_apt_get_update
+
+	do_with_retries 3 chroot_sdcard "git clone --depth=1 https://github.com/andersson/qrtr.git /root/qrtr"
+	do_with_retries 3 chroot_sdcard "git clone --depth=1 https://github.com/andersson/pd-mapper.git /root/pd-mapper"
+	do_with_retries 3 chroot_sdcard "git clone --depth=1 https://github.com/andersson/tqftpserv.git /root/tqftpserv"
+	do_with_retries 3 chroot_sdcard "git clone --depth=1 https://github.com/ericcurtin/qbootctl.git /root/qbootctl"
+	chroot_sdcard "pushd /root/qrtr && make prefix=/usr && make prefix=/usr install && systemctl enable qrtr-ns && popd && rm -rf /root/qrtr"
+	chroot_sdcard "pushd /root/pd-mapper && make prefix=/usr && make prefix=/usr install && systemctl enable pd-mapper && popd && rm -rf /root/pd-mapper"
+	chroot_sdcard "pushd /root/tqftpserv && make prefix=/usr && make prefix=/usr install && systemctl enable tqftpserv && popd && rm -rf /root/tqftpserv"
+	chroot_sdcard "pushd /root/qbootctl && meson build && meson compile -C build && chmod +x build/qbootctl && mv build/qbootctl /usr/bin/ && popd && rm -rf /root/qbootctl"
+
+	chroot_sdcard "apt purge make gcc g++ liblzma-dev zlib1g-dev meson cmake -y"
```

Comment:
Is it necessary to build these from source. Ubuntu noble should aready have them.

Reply:
I'm not sure if Noble's mesa includes zink, I'll test the rest later